File: set 6 - Python/mario-more/mario.py
# calls the get_int funcntion from cs50 library
from cs50 import get_int

# main method of the program


def main():
    height = get_int_onRange(1, 8)
    build_pyramid(height)

# deals with prompting int, range min-max inclusive


def get_int_onRange(min, max):
    while True:
        n = get_int("Height: ")
        if n >= min and n <= max:
            break
    return n

# contruct the pyramid of the given height


def build_pyramid(height):
    for i in range(height):
        for j in range(height - i - 1):
            print(" ", end="")
        put_block(i)
        print("  ", end="")
        put_block(i)
        print()

# provides n blocks to put in the line


def put_block(n):
    for i in range(n + 1):
        print("#", end="")


# calls the main function
main()

File: set 1 - C/mario-less/mario.c
#include <stdio.h>
#include <cs50.h>

// abstract method that deals with prompting int, range min-max inclusive
int get_int_onRange(int min, int max);

// abstract method contruct the pyramid of the given height
void build_pyramid(int height);

int main(void)
{
    int height = get_int_onRange(1, 8);
    build_pyramid(height);
}

// method implements get_int_onRange
int get_int_onRange(int min, int max)
{
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n < min || n > max);
    return n;
}

// method implements get_int_onRange
void build_pyramid(int height)
{
    for (int i = 0; i < height; i++)
    {
        for (int j = 1; j < height - i; j++)
        {
            printf(" ");
        }
        for (int k = 0; k < i + 1; k++)
        {
            printf("#");
        }
        printf("\n");
    }
}